File: binary-tree-general/invert-binary-tree/invert-binary-tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution
{
    public TreeNode InvertTreeDfs(TreeNode root)
    {
        if (root == null) return null;

        TreeNode left = InvertTree(root.left);
        TreeNode right = InvertTree(root.right);

        root.left = right;
        root.right = left;

        return root;
    }
    
    public TreeNode InvertTreeBfsQueue(TreeNode root) {
        if (root == null) return null;

        var queue = new Queue<TreeNode>();
        queue.Enqueue(root);

        while (queue.Count > 0) {
            var node = queue.Dequeue();

            (node.left, node.right) = (node.right, node.left);

            if (node.left != null) queue.Enqueue(node.left);
            if (node.right != null) queue.Enqueue(node.right);
        }

        return root;
    }
}

File: binary-tree-general/invert-binary-tree/invert-binary-tree.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function invertTreeDfs(root: TreeNode | null): TreeNode | null {
    if (!root) return null;

    const left = invertTreeDfs(root.left);
    const right = invertTreeDfs(root.right);

    root.left = right;
    root.right = left;

    return root;
};

function invertTreeBfsQueue(root: TreeNode | null): TreeNode | null {
    if (!root) return null;

    const queue: TreeNode[] = [root];

    while (queue.length > 0) {
        const node = queue.shift()!;

        [node.left, node.right] = [node.right, node.left];

        if (node.left) queue.push(node.left);
        if (node.right) queue.push(node.right);
    }

    return root;
}